src/imgtools/cli/autopipeline.py

```
@@ -0,0 +1,245 @@
+        roi_on_missing_regex=ROIMatchFailurePolicy[roi_on_missing_regex],
+        spacing=spacing,
+        window=window,
+        level=level,
+    )
+    
+    # Run the pipeline
+    try:
+        results = pipeline.run(first_n=first_n)
+    except Exception as e:
+        logger.exception(f"Error running pipeline: {str(e)}")
+        raise click.Abort()
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)